operator_ui/src/pages/Jobs/Show.tsx

```diff
@@ -77,22 +77,45 @@ export const JobsShow: React.FC<Props> = ({ match }) => {
 
   const getJobSpec = React.useCallback(async () => {
     if (isOcrJob(jobSpecId)) {
-      return v2.ocrSpecs
+      return v2.jobs
         .getJobSpec(jobSpecId)
         .then((response) => {
           const jobSpec = response.data
-          setState((s) => ({
-            ...s,
-            jobSpec,
-            job: {
-              ...jobSpec.attributes,
-              ...jobSpec.attributes.offChainReportingOracleSpec,
+          setState((s) => {
+            const job = {
               ...jobSpec.attributes.pipelineSpec,
               id: jobSpec.id,
               definition: generateTOMLDefinition(jobSpec.attributes),
-              type: 'Off-chain reporting',
-            },
-          }))
+              type: 'v2',
+              name: jobSpec.attributes.name,
+              specType: jobSpec.attributes.type as JobSpecType,
+              errors: jobSpec.attributes.errors,
+            } as JobV2
+
+            switch (jobSpec.attributes.type) {
+              case 'offchainreporting':
+                job.createdAt = jobSpec.attributes.offChainReportingOracleSpec
+                  ?.createdAt as string
```

Comment:
I believe this is resolved in that it removes the optional chaining however I believe we still need the switch, because the object just represents the API response, and you need to know the type to get the `createdAt` from the correct spec

Reply:
I think this would be a better way to do this https://github.com/smartcontractkit/chainlink/pull/3981#discussion_r588100217